hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/filtermanagement/AutoAssignmentWindowController.java

```diff
@@ -112,16 +130,17 @@ protected void persistEntity(final ProxyTargetFilterQuery entity) {
     }
 
     private void showConsequencesDialog(final String confirmationCaption, final String confirmationQuestion,
-            final Long targetFilterId, final Long autoAssignDsId, final ActionType autoAssignActionType) {
-        final ConfirmationDialog confirmDialog = new ConfirmationDialog(getI18n(), confirmationCaption,
-                confirmationQuestion, ok -> {
-                    if (ok) {
-                        final TargetFilterQuery targetFilterQuery = targetFilterQueryManagement.updateAutoAssignDS(
-                                getEntityFactory().targetFilterQuery().updateAutoAssign(targetFilterId)
-                                        .ds(autoAssignDsId).actionType(autoAssignActionType));
-                        publishModifiedEvent(createModifiedEventPayload(targetFilterQuery));
-                    }
-                }, UIComponentIdProvider.DIST_SET_SELECT_CONS_WINDOW_ID);
+            final String confirmationHint, final Long targetFilterId, final Long autoAssignDsId,
+            final ActionType autoAssignActionType, final boolean confirmationRequired) {
+        final ConfirmationDialog confirmDialog = ConfirmationDialog
+                .newBuilder(getI18n(), UIComponentIdProvider.DIST_SET_SELECT_CONS_WINDOW_ID)
+                .caption(confirmationCaption).question(confirmationQuestion).hint(confirmationHint)
```

Comment:
Just wondering if this should be called as "hint" or as a "warning" or something that tells user to be aware of the future consequences

Reply:
Since we are not adding some warning icon here and only attach some text, I thought hint would be enough here. Let's discuss it in the feature demo.